sbt-metals/src/main/scala/scala/meta/sbt/MetalsPlugin.scala

```diff
@@ -0,0 +1,189 @@
+package scala.meta.sbt
+
+import sbt._
+import sbt.Keys._
+import java.io._
+
+object MetalsPlugin extends AutoPlugin {
+  override def trigger = allRequirements
+  override def requires = sbt.plugins.JvmPlugin
+  object autoImport {
+
+    val metalsBuildInfo =
+      taskKey[Map[String, String]](
+        "List of key/value pairs for build information such as classpath/sourceDirectories"
+      )
+    val metalsWriteBuildInfo =
+      taskKey[Unit](
+        "Write build information to .metals/buildinfo/"
+      )
+
+    lazy val semanticdbSettings = Seq(
+      addCompilerPlugin(
+        "org.scalameta" % "semanticdb-scalac" % Metals.semanticdbVersion cross CrossVersion.full
+      ),
+      scalacOptions += "-Yrangepos"
+    )
+
+    def metalsSettings(cs: Configuration*): Seq[Def.Setting[_]] = {
+      val configs = if (cs.nonEmpty) cs else Seq(Compile)
+      configs.flatMap { config =>
+        inConfig(config)(
+          Seq(
+            metalsBuildInfo := Metals.metalsBuildInfoTask.value,
+            metalsWriteBuildInfo := Metals.metalsWriteBuildInfoTask.value,
+          )
+        )
+      } ++ Seq(
+        // without config scope it will aggregate over all project dependencies
+        // and their configurations
+        metalsWriteBuildInfo := Def.taskDyn {
+          val depsAndConfigs = ScopeFilter(
+            inDependencies(ThisProject),
+            inConfigurations(configs: _*)
+          )
+          metalsWriteBuildInfo.all(depsAndConfigs)
+        }.value
+      )
+    }
+  }
+  import autoImport._
+
+  override def projectSettings =
+    metalsSettings(Compile, Test)
+
+  override def globalSettings = Seq(
+    commands ++= Seq(
+      Metals.semanticdbEnable,
+      Metals.metalsSetup,
+    ),
+    // without project scope it will aggregate over all projects
+    metalsWriteBuildInfo := {
+      metalsWriteBuildInfo.all(ScopeFilter(inAnyProject)).value
+      streams.value.log.info("Metals rocks! 🤘")
```

Comment:
😆

Reply:
We already have some emoji in the log from `semanticdbEnable`, so I thought, why not? I hope it will make some users smile when they first try Metals... 😊

Reply:
I like emojis 👍 

How about going all in and making it more consistent? 😄 


> 👌 semanticdb-scalac enabled
> 🔨 created .metals/buildinfo/foo/test.properties
> 🤘 Metals is ready, time to rock!